src/ServiceProvider.php

```diff
@@ -158,4 +175,17 @@ protected function loadPublishing()
             $this->publishes([$viewsSource => resource_path('views/vendor/service')], 'views');
         }
     }
+
+    public function loadJobProcessEvents()
```

Comment:
Should this perhaps only be done when being run in cli?

The scenario I can see being troublesome is:

* Handle request, audit-log in controller/mutation, new correlation id is generated.
* Dispatch job (using `WithCorrelationId`) to a sync queue. Job is immediately processed and the correlation Id is set.
* Correlation id is reset after the job has been processed.
* Subsequent audit-logs (in jobs, controller, middleware etc) in the same request will get a new correlation id.

Some possible ways around it:

* Reset the correlationI d to its previous value in `Queue::after` instead of resetting it to null. If it was null before it will be null after but if it already had a correlation id that correlation id will be reset.
* Only call ` $this->loadJobProcessEvents();` when the application is run in cli-mode.
* Don't call ` $this->loadJobProcessEvents();` when the queue is in sync-mode.

Reply:
Good call.
I went with the "Only call `$this->loadJobProcessEvents()` when the application is run in cli-mode." solution.